pkg/agent/controller/networkpolicy/audit_logging_test.go

```diff
@@ -65,57 +72,81 @@ func newLogInfo(disposition string) (*logInfo, string) {
 	}, expected
 }
 
-func sendMultiplePackets(antreaLogger *AntreaPolicyLogger, ob *logInfo, numPackets int) {
+func sendMultiplePackets(antreaLogger *AntreaPolicyLogger, ob *logInfo, numPackets int, sendInterval time.Duration) {
 	count := 0
-	for range time.Tick(12 * time.Millisecond) {
+	for range time.Tick(sendInterval) {
 		count += 1
-		antreaLogger.logDedupPacket(ob, 100*time.Millisecond)
+		antreaLogger.LogDedupPacket(ob)
 		if count == numPackets {
 			break
 		}
 	}
 }
 
+func closePacketTransmit(mockAnpLogger *mockLogger, testWaitTime time.Duration) {
+	time.Sleep(testWaitTime)
+	mockAnpLogger.mu.Lock()
+	defer mockAnpLogger.mu.Unlock()
+	close(mockAnpLogger.logged)
+}
+
 func expectedLogWithCount(msg string, count int) string {
 	return fmt.Sprintf("%s [%d", msg, count)
 }
 
 func TestAllowPacketLog(t *testing.T) {
-	antreaLogger, mockAnpLogger := newTestAntreaPolicyLogger()
+	antreaLogger, mockAnpLogger := newTestAntreaPolicyLogger(testBufferLength)
 	ob, expected := newLogInfo("Allow")
 
-	antreaLogger.logDedupPacket(ob, 100*time.Millisecond)
+	antreaLogger.LogDedupPacket(ob)
 	actual := <-mockAnpLogger.logged
 	assert.Contains(t, actual, expected)
 }
 
 func TestDropPacketLog(t *testing.T) {
-	antreaLogger, mockAnpLogger := newTestAntreaPolicyLogger()
+	antreaLogger, mockAnpLogger := newTestAntreaPolicyLogger(testBufferLength)
 	ob, expected := newLogInfo("Drop")
 
-	antreaLogger.logDedupPacket(ob, 100*time.Millisecond)
+	antreaLogger.LogDedupPacket(ob)
 	actual := <-mockAnpLogger.logged
 	assert.Contains(t, actual, expected)
 }
 
 func TestDropPacketDedupLog(t *testing.T) {
-	antreaLogger, mockAnpLogger := newTestAntreaPolicyLogger()
+	antreaLogger, mockAnpLogger := newTestAntreaPolicyLogger(testBufferLength)
 	ob, expected := newLogInfo("Drop")
-	// add the additional log info for duplicate packets
+	// Add the additional log info for duplicate packets.
 	expected = expectedLogWithCount(expected, 2)
 
-	go sendMultiplePackets(antreaLogger, ob, 2)
+	go sendMultiplePackets(antreaLogger, ob, 2, time.Millisecond)
 	actual := <-mockAnpLogger.logged
 	assert.Contains(t, actual, expected)
 }
 
 func TestDropPacketMultiDedupLog(t *testing.T) {
-	antreaLogger, mockAnpLogger := newTestAntreaPolicyLogger()
+	antreaLogger, mockAnpLogger := newTestAntreaPolicyLogger(testBufferLength)
 	ob, expected := newLogInfo("Drop")
 
-	go sendMultiplePackets(antreaLogger, ob, 10)
-	actual := <-mockAnpLogger.logged
-	assert.Contains(t, actual, expectedLogWithCount(expected, 9))
-	actual = <-mockAnpLogger.logged
-	assert.Contains(t, actual, expected)
+	numPackets := 4
+	go sendMultiplePackets(antreaLogger, ob, numPackets, 40*time.Millisecond)
+	// Close the channel listening for logged msg after 1s.
+	go closePacketTransmit(mockAnpLogger, 500*time.Millisecond)
```

Comment:
you can use `time.AfterFunc(500*time.Millisecond, func() { close(mockAnpLogger.logged) })` for this I believe